src/index.js

```diff
@@ -26,22 +27,40 @@ async function run() {
         throw new Error('The changelog-scopes input cannot be parsed as JSON.');
     }
 
-    let gitLog;
-    // Fetch base and head branches with history and git message log diff
-    if (compareMethod === 'branch') {
-        await exec(`git fetch origin ${baseBranch} ${branch}`);
-        ({ stdout: gitLog } = await exec(`git log --no-merges --pretty='%s' origin/${branch} ^origin/${baseBranch}`));
-    } else if (compareMethod === 'tag') {
-        // fetch base branch and get commit history from latest tag. If tag is not found fetch whole history.
-        await exec(`git fetch origin ${baseBranch}`);
-        const { stdout: tag } = await exec(`git describe --tags --abbrev=0`);
-        const start = tag ? `${tag.replace(/[\r\n]/gm, '')}..` : '';
-        ({ stdout: gitLog } = await exec(`git log --no-merges --pretty='%s' ${start}HEAD`));
+    let gitMessages;
+    if (compareMethod === 'pull_request') {
+        // Get PR number
+        const eventFileContent = await fs.readFile(process.env.GITHUB_EVENT_PATH);
+        const prNumber = JSON.parse(eventFileContent).pull_request.number;
+
+        if (!prNumber) throw new Error('Could not obtain pull request\'s number. Was the workflow trigger "pull_request"?');
+        const gitLog = await repoOctokit.rest.pulls.listCommits({
+            owner: github.context.repo.owner,
+            repo: github.context.repo.repo,
+            pull_number: prNumber,
+        });
+        gitMessages = gitLog.data.map((commit) => commit.commit.message);
     } else {
-        throw new Error(`Unrecognized "compare-method" value: ${compareMethod}`);
+        let gitLog;
+        // Fetch base and head branches with history and git message log diff
+        if (compareMethod === 'branch') {
+            await exec(`git fetch origin ${baseBranch} ${branch}`);
+            ({ stdout: gitLog } = await exec(`git log --no-merges --pretty='%s' origin/${branch} ^origin/${baseBranch}`));
```

Comment:
Yeah I thought about it as well, but I wanted to avoid breaking changes for now. This PR is meant more like a hotfix and  though I tested this new way of fetching commits on dummy repo, I'd wait after this is proved to be bullet-proofed before rewriting methods that work for now.